Sources/MapboxMaps/Ornaments/Compass/MapboxCompassOrnamentView.swift

```diff
@@ -51,19 +51,25 @@ internal class MapboxCompassOrnamentView: UIButton {
                                               comment: "Accessibility hint")
 
         if let image = createCompassImage() {
-            containerView.image = image
-            NSLayoutConstraint.activate([
-                widthAnchor.constraint(equalToConstant: image.size.width),
-                heightAnchor.constraint(equalToConstant: image.size.height),
-                containerView.widthAnchor.constraint(equalToConstant: image.size.width),
-                containerView.heightAnchor.constraint(equalToConstant: image.size.height)
-            ])
+            updateImage(image: image)
         }
         containerView.translatesAutoresizingMaskIntoConstraints = false
         addSubview(containerView)
         addTarget(self, action: #selector(didTap), for: .touchUpInside)
     }
 
+    func updateImage(image: UIImage?) {
+        if image != nil {
+            containerView.image = image
+            NSLayoutConstraint.activate([
```

Comment:
Old constraints needs to be deactivated, as there will be conflicting constraints if image size changed.

Alternatively, the container view can be constrained to the edges of the compass ornament view. You could store references to the width/height constraints of the compass ornament view and update them when updating the image.

```swift
widthConstraint.constant = image.size.width
heightConstraint.constant = image.size.height
```

Reply:
interesting, I thought if we constraint the subview to the edges of its superview and set some compression/hugging priority then the superview will also resize to always fit the subview?

Reply:
was able to validate it doesn't throw any warnings anymore about constraints!